src/map/LayerHelper.js

```diff
@@ -296,17 +303,25 @@ exports.prototype.getFlatLayers = function(layer) {
  * of others groups.
  * @param {ol.layer.Base} layer The base layer, mostly a group of layers.
  * @param {Array.<ol.layer.Base>} array An array to add layers.
+ * @param {number|undefined} computedOpacity Opacity inherited from ancestor layer groups.
  * @return {Array.<ol.layer.Layer>} Layers.
  * @private
  */
-exports.prototype.getFlatLayers_ = function(layer, array) {
-  if (layer instanceof olLayerGroup) {
+exports.prototype.getFlatLayers_ = function(layer, array, computedOpacity) {
+  const opacity = layer.getOpacity();
+  if (computedOpacity !== undefined) {
+    computedOpacity *= opacity;
+  } else {
+    computedOpacity = opacity;
+  }
+  if (layer instanceof OlLayerGroup) {
     const sublayers = layer.getLayers();
     sublayers.forEach((l) => {
-      this.getFlatLayers_(l, array);
+      this.getFlatLayers_(l, array, computedOpacity);
     });
   } else {
     if (array.indexOf(layer) < 0) {
+      layer.set('inheritedOpacity', computedOpacity, true);
```

Comment:
Can you check in our code if there is a way to indicate (document) that this opacity and should be used ?